drivers/flash/Kconfig.stm32

```
@@ -6,6 +6,15 @@
 # Copyright (c) 2023 Google Inc
 # SPDX-License-Identifier: Apache-2.0
 
+config STM32_MEMMAP
+	bool "NOR Flash in MemoryMapped for XiP"
+	depends on XIP && \
+		    (DT_HAS_ST_STM32_OSPI_NOR_ENABLED || \
+		     DT_HAS_ST_STM32_QSPI_NOR_ENABLED)
```

To be confirmed by @FRASTM, but I think it should be
```suggestion
		    (DT_HAS_ST_STM32_XSPI_NOR_ENABLED || \
		     DT_HAS_ST_STM32_OSPI_NOR_ENABLED || \
		     DT_HAS_ST_STM32_QSPI_NOR_ENABLED)
```

---

It doesn't look like the XSPI driver supports memory map mode.

---

> It doesn't look like the XSPI driver supports memory map mode.

Not yet, that is coming soon ...

---

@FRASTM, should `DT_HAS_ST_STM32_XSPI_NOR_ENABLED` be added as a dependency?